guvnor-ala/guvnor-ala-registry-vfs/src/main/java/org/guvnor/ala/registry/vfs/VFSPipelineExecutorRegistry.java

```diff
@@ -0,0 +1,116 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.registry.vfs;
+
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.guvnor.ala.pipeline.execution.PipelineExecutorTrace;
+import org.guvnor.ala.registry.impl.BasePipelineExecutorRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.java.nio.file.Path;
+
+import static org.guvnor.ala.registry.vfs.VFSRegistryHelper.BySuffixFilter.newFilter;
+import static org.uberfire.commons.validation.PortablePreconditions.checkNotNull;
+
+/**
+ * VFS based implementation of a PipelineExecutorRegistry.
+ */
+@ApplicationScoped
+public class VFSPipelineExecutorRegistry
+        extends BasePipelineExecutorRegistry {
+
+    protected static final String PIPELINE_EXECUTOR_REGISTRY_PATH = "executor-registry";
+
+    protected static final String TRACE_SUFFIX = "-trace.entry";
+
+    private static final Logger logger = LoggerFactory.getLogger(VFSPipelineExecutorRegistry.class);
+
+    private VFSRegistryHelper registryHelper;
+
+    private org.uberfire.java.nio.file.Path registryRoot;
+
+    public VFSPipelineExecutorRegistry() {
+        //Empty constructor for Weld proxying
+    }
+
+    @Inject
+    public VFSPipelineExecutorRegistry(final VFSRegistryHelper registryHelper) {
+        this.registryHelper = registryHelper;
+    }
+
+    @PostConstruct
+    public void init() {
+        initializeRegistryRoot();
+        initializeRegistry();
+    }
+
+    @Override
+    public void register(final PipelineExecutorTrace trace) {
+        checkNotNull("trace",
+                     trace);
+        final Path path = buildTracePath(trace.getTaskId());
+        try {
+            registryHelper.storeEntry(path,
+                                      trace);
+        } catch (Exception e) {
+            //uncommon error
+            logger.error("Unexpected error was produced during trace marshalling/storing, trace: " + trace,
+                         e);
+            throw new RuntimeException("Unexpected error was produced during trace marshalling/storing, trace: " + trace,
+                                       e);
+        }
+        super.register(trace);
+    }
+
+    public void deregister(final String taskId) {
+        checkNotNull("taskId",
+                     taskId);
+        final Path path = buildTracePath(taskId);
+        registryHelper.deleteBatch(path);
+        super.deregister(taskId);
+    }
+
+    private void initializeRegistryRoot() {
+        try {
+            registryRoot = registryHelper.ensureDirectory(PIPELINE_EXECUTOR_REGISTRY_PATH);
+        } catch (Exception e) {
+            //uncommon error
+            logger.error("An error was produced during " + VFSPipelineExecutorRegistry.class.getName() +
```

Comment:
I think the same I've tried to explain above. The failure in the initialization case is a sort of uncommon failure that occurs 99% in cases where the VFS wasn't initialized properly. And all the WF will be producing countless errors. So I believe that in the initialization case there's not much to do.

Assuming the initialization was ok, and the WB, etc., is working well, when things starts to be registered, I might happen the case where a particular registering can't be produced, and in this case it makes sense to catch, log the error, and re-throw the exception